Fix part of #10616: Introduced 'typescript-eslint/dot-notation' rule

goldberl opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #10616 .
2. This PR does the following: added typescript rules for `@typescript-eslint/dot-notation` to the `eslintrc` file.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

Linter checks pass when code is pushed up:
![image](https://user-images.githubusercontent.com/76408777/222013470-3262b218-4d70-4a22-b349-e049e8455f8c.png)

I will note that `python -m scripts.linters.pre_commit_linter --only-check-file-extensions=js --path=. --verbose` does not work right now and I have [commented about it here on the issue](https://github.com/oppia/oppia/issues/10616#issuecomment-1447376921) and [opened an issue for it here](https://github.com/oppia/oppia/issues/17472).

All frontend tests pass:
![image](https://user-images.githubusercontent.com/76408777/222016089-20733e09-96cf-400d-9061-4a5e829b63e5.png)

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

Commit:
added typescript eslint dot notation rule

Commit:
Merge branch 'develop' into ts_eslint_dotnot

goldberl committed:
Merge branch 'oppia:develop' into ts_eslint_dotnot

Commit:
Merge branch 'develop' into ts_eslint_dotnot

goldberl committed:
Merge branch 'oppia:develop' into ts_eslint_dotnot

gp201 commented:
@goldberl The lint check seems to be failing PTAL. Thanks!

Commit:
Merge branch 'develop' into ts_eslint_dotnot

Commit:
Merge branch 'ts_eslint_dotnot' of https://github.com/goldberl/oppia into ts_eslint_dotnot

Commit:
added project property to parserOptions

Commit:
added comma

goldberl committed:
Merge branch 'develop' into ts_eslint_dotnot

FutzMonitor commented:
Is there any way to run only the tests that are failing?

FutzMonitor commented:
I notice this repeated wording in the failed lint tests
```
0:0  error  Parsing error: ESLint was configured to run on 
`<tsconfigRootDir>/core/templates/base-components/warnings-and-alerts.component.spec.ts`
using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
```
Could there be an issue with configurations someplace? 
Maybe give this a try [StackOverFlow](https://stackoverflow.com/questions/74413483/eslint-cypress-parsing-error-eslint-was-configured-to-run-on-tsconfigrootd)

goldberl commented:
> I notice this repeated wording in the failed lint tests
> 
> ```
> 0:0  error  Parsing error: ESLint was configured to run on 
> `<tsconfigRootDir>/core/templates/base-components/warnings-and-alerts.component.spec.ts`
> using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
> However, that TSConfig does not include this file. Either:
> - Change ESLint's list of included files to not include this file
> - Change that TSConfig to include this file
> - Create a new TSConfig that includes this file and include it in your parserOptions.project
> ```
> 
> Could there be an issue with configurations someplace?

@FutzMonitor There is likely an issue with value of the `project` property that I added to the `.eslintrc` file. Before, I added that, I got this error:
![image](https://user-images.githubusercontent.com/76408777/222577065-c87d9f74-7461-40b9-a33d-ee2f16a68136.png)

And so I added a `project` property with the value `tsconfig.json` based on the documentation of the `typescript-eslint/parser` rule found[ here](https://typescript-eslint.io/architecture/parser/#project). This value is probably what is causing the current errors. I am not quite sure what to change it to so that these errors do not occur. @gp201 If you have any ideas on this, please let me know when you get a chance, thanks!

FutzMonitor commented:
> There is likely an issue with value of the `project` property that I added to the `.eslintrc` file. Before, I added that, I got this error: ![image](https://user-images.githubusercontent.com/76408777/222577065-c87d9f74-7461-40b9-a33d-ee2f16a68136.png)
> 
> And so I added a `project` property with the value `tsconfig.json` based on the documentation of the `typescript-eslint/parser` rule found[ here](https://typescript-eslint.io/architecture/parser/#project). This value is probably what is causing the current errors. I am not quite sure what to change it to so that these errors do not occur.

I was reading this GitHub conversation [here](https://github.com/nestjs/nest/issues/4900), maybe there are some potential solutions here for you to try out.

goldberl committed:
Merge branch 'oppia:develop' into ts_eslint_dotnot

Commit:
edited project property

Commit:
Merge branch 'develop' into ts_eslint_dotnot

Commit:
Merge branch 'develop' into ts_eslint_dotnot

Commit:
Merge branch 'ts_eslint_dotnot' of https://github.com/goldberl/oppia into ts_eslint_dotnot

goldberl committed:
Merge branch 'oppia:develop' into ts_eslint_dotnot

Commit:
Merge branch 'develop' into ts_eslint_dotnot

goldberl committed:
Merge branch 'develop' into ts_eslint_dotnot

goldberl committed:
Merge branch 'develop' into ts_eslint_dotnot

Commit:
Merge branch 'develop' into ts_eslint_dotnot

goldberl committed:
Merge branch 'oppia:develop' into ts_eslint_dotnot

Commit:
Merge branch 'ts_eslint_dotnot' of https://github.com/goldberl/oppia into ts_eslint_dotnot

Commit:
Merge branch 'develop' into ts_eslint_dotnot

goldberl committed:
Merge branch 'develop' into ts_eslint_dotnot

Commit:
edited parser project property

Commit:
Merge branch 'develop' into ts_eslint_dotnot

Commit:
Merge branch 'ts_eslint_dotnot' of https://github.com/goldberl/oppia into ts_eslint_dotnot

goldberl committed:
Merge branch 'oppia:develop' into ts_eslint_dotnot

Commit:
Merge branch 'ts_eslint_dotnot' of https://github.com/goldberl/oppia into ts_eslint_dotnot
